src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php

```diff
@@ -65,4 +59,28 @@ protected function isRegex($str)
      * @return string regexp corresponding to a given string
      */
     abstract protected function toRegex($str);
+
+    private function buildRegexps($patterns)
+    {
+        $rxs = array();
+        $regexps = array();
+
+        foreach ($patterns as $pattern) {
+            $regex = Regex::create($this->toRegex($pattern))->render();
+
+            if (preg_match('/\W([imsxUXJ]*)$/', $regex, $match)) {
+                $rxs[] = '(?'.$match[1].':'.substr($regex, 1, -strlen($match[0])).')';
+            } else {
+                $regexps[] = $regex;
+            }
+        }
+        if (1 < count($rxs)) {
+            $regexps[] = Regex::BOUNDARY.implode('|', $rxs).Regex::BOUNDARY;
+        } elseif ($rxs) {
+            $rxs = explode(':', substr($rxs[0], 2, -1), 2);
+            $regexps[] = Regex::BOUNDARY.$rxs[1].Regex::BOUNDARY.$rxs[0];
+        }
+
+        return $regexps;
+    }
```

Comment:
OK, removed. Just in case and for history, I'll open an other PR with this patch, with a fix for excluding unmergeable regexps.